quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/PropertyMappingInterceptor.java

```diff
@@ -105,10 +108,20 @@ public Iterator<String> iterateNames(ConfigSourceInterceptorContext context) {
             }
             allMappers.remove(mapper);
 
-            // include additional mappings if we're on the from side of the mapping
-            // as the mapping may not be bi-directional
+            if (mapper.hasWildcard()) {
+                // non-wildcard connected options should be already advertised by the logic in iterateNames()
+                if (mapper.hasConnectedOptions()) {
+                    var wildcardMapper = (WildcardPropertyMapper<?>) mapper;
+                    var wildcardValue = wildcardMapper.extractWildcardValue(name).orElseThrow();
+                    var connectedTo = wildcardMapper.getConnectedOptions(wildcardValue).stream()
+                            .map(option -> Optional.ofNullable(PropertyMappers.getMapper(NS_KEYCLOAK_PREFIX + option)).orElseThrow(() -> new IllegalArgumentException("Cannot find connected options")))
+                            .map(m -> m.hasWildcard() ? ((WildcardPropertyMapper<?>) m).getTo(wildcardValue) : m.getTo());
 
-            if (!mapper.hasWildcard() && name.equals(mapper.getFrom())) {
+                    return Stream.concat(toDistinctStream(name, wildcardMapper.getTo(wildcardValue)), connectedTo);
+                }
+            } else if (name.equals(mapper.getFrom())) {
+                // include additional mappings if we're on the from side of the mapping as the mapping may not be bi-directional
```

Comment:
This should apply in the wildcard case as well - if the user puts the property directly in quarkus.properties, then we don't need to take responsibility for advertising the keycloak option.

Reply:
Makes sense.